content_handler/runtime_holder.cc

```diff
@@ -448,9 +454,19 @@ void RuntimeHolder::InitMozartInternal() {
 }
 
 void RuntimeHolder::InitRootBundle(std::vector<char> bundle) {
-  root_bundle_data_ = std::move(bundle);
-  asset_store_ = fxl::MakeRefCounted<blink::ZipAssetStore>(
-      GetUnzipperProviderForRootBundle());
+  if (!bundle.empty()) {
+    root_bundle_data_ = std::move(bundle);
+    asset_store_ = fxl::MakeRefCounted<blink::ZipAssetStore>(
+        GetUnzipperProviderForRootBundle());
+  } else {
+    fxl::UniqueFD root_dir(fdio_ns_opendir(namespc_));
+    if (!root_dir.is_valid()) {
+      FXL_LOG(ERROR) << "Unable to load root dir";
+      return;
+    }
+    directory_asset_bundle_ = fxl::MakeRefCounted<blink::DirectoryAssetBundle>(
+        std::move(root_dir), kFuchsiaPackageResourceDirectory);
```

Comment:
Why not open kFuchsiaPackageResourceDirectory and pass that fd to DirectoryAssetBundle?  Then it can just work from that point.